internal/hvsocket/hvsocket.go

```diff
@@ -0,0 +1,81 @@
+//go:build windows
+// +build windows
+
+package hvsocket
+
+import (
+	"context"
+	"fmt"
+	"unsafe"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+	"golang.org/x/sys/windows"
+
+	"github.com/Microsoft/hcsshim/internal/resources"
+)
+
+const (
+	addressFlagPassthru            = 0x00000001
+	ioCtlHVSocketUpdateAddressInfo = 0x21c004
```

Comment:
nit: Is this a publicly document windows IOCTL? If yes, can we add a link to the documentation here?

Reply:
Hmm, in that case can we add a small comment either here (or in a doc.go file for this package) explaining what this IOCTL is doing, how we plan to use it and how the `CreateAddressInfo` function should be used?

Reply:
did some refactoring and added a docstring.